SparkPost: Add template-based message support and basic sending-domain registration functions

nickvanderveeken opened the pull request.

roelvanduijnhoven commented on src/Mail/Message/Converter/SparkPostConverter.php:
```diff
@@ -0,0 +1,58 @@
+use SlmMail\Mail\Message\SparkPost;
+use SlmMail\Mail\Message\Mandrill;
+
+class SparkPostConverter
```
@nickvanderveeken I would place this code inside our _own_ repo TBH!

roelvanduijnhoven commented on src/Service/SparkPostService.php:
```diff
@@ -131,93 +203,220 @@ protected function prepareRecipients(Message $message): array
         return $message->getBodyText();
     }
 
+    public function registerSendingDomain(string $domain): bool
```
@nickvanderveeken I think it makes more sense to simply pass the DKIM parameter as (optional) arguments to this method. I think in general the DKIM settings belong to this method call; and there is no need for a _default_ DKIM config. 

I could be missing something here; so please let me know if that is the case!

nickvanderveeken replied:
Yeah, it makes more sense.

roelvanduijnhoven commented on src/Service/SparkPostService.php:
```diff
@@ -131,93 +203,220 @@ protected function prepareRecipients(Message $message): array
      * @return HttpClient
      */
-    private function prepareHttpClient(string $uri, array $parameters = []): HttpClient
+    protected function prepareHttpClient(string $uri, array $parameters = []): HttpClient
```
Is there a use case for overriding this? I would think that _if_ you simply inject another client to the constructor of this service; to augment or extend the client itself.

roelvanduijnhoven reviewed (commented):
Thanks for the contributon :). Nice to see the Sparkpost grow in it's supported features!

Some remarks on how I think we can improve DKIM.

Let me know once I can merge this thing; and release a new version.

nickvanderveeken commented:
@roelvanduijnhoven I addressed your comments, adjusted the PR and succesfully integration-tested it with my application code. I'd say it's good to merge it now.

roelvanduijnhoven commented:
@nickvanderveeken Launched version v4.1 :). Thanks a lot!
